types/types.go

```diff
@@ -320,3 +320,21 @@ type FeeHistoryResult struct {
 	BaseFee      []*hexutil.Big   `json:"baseFeePerGas,omitempty"`
 	GasUsedRatio []float64        `json:"gasUsedRatio"`
 }
+
+type TraceTransactionResult struct {
```

Comment:
It seems that some fields (`Failed`) are missing and some types (`Depth`, `ReturnValue`) are slightly different from geth: https://github.com/ethereum/go-ethereum/blob/master/eth/tracers/logger/logger.go#L406

Is it possible to use geth's `ExecutionResult` directly? Or is the l2geth return type different?

Reply:
yep. I didn't know this previously.